src/coin_test/data/datasets.py

```diff
@@ -80,6 +80,89 @@ def process(self, processors: list[Processor]) -> "Dataset":
         self.df = df
         return self  # Return self for caller's convenience
 
+    @staticmethod
+    def _calculate_split_index(
+        dataset: "Dataset",
+        timestamp: pd.Timestamp | None = None,
+        length: pd.Timedelta | None = None,
```

Comment:
We should try to take `pd.Timedelta | pd.DateOffset`. Both are valid ways of specifying length, but one is exact and one respects calendar irregularities. It should be a simple change - only the type annotation needs an addition.